v1/proto/service/gribi.proto

```diff
@@ -421,3 +427,49 @@ message GetResponse {
   // A set of entries that are installed on the device.
   repeated AFTEntry entry = 1;
 }
+
+// FlushRequest is the request sent from the client to the server when a Flush
+// RPC is called.
+message FlushRequest {
+  // election specifies the election ID from a client. It is valid only
+  // when the server has clients that are operating in the SINGLE_PRIMARY mode.
+  oneof election {
+    // id indicates that the Flush ID supplied should be compared to the
+    // existing election ID on the server from the current master.
```

Comment:
should we add more text here. I'm guessing the `id` needs to be equal to the primary's election id for the server to accept the request. What if it's > than the primary's? Is that valid?

Note that comparing against current primary means that if the primary is disconnected or there are no ModifyRPC streams, even though the server knows the previous highest election_id it'll reject the flush request.
In other words, flush will be honored if id == primary's election_id and primary is connected.

Reply:
Thanks for catching this -- yes, definitely agree that there is some work here to be clearer, how does the following text sound: 

```
    // id indicates that the Flush ID supplied should be compared to the
    // existing election ID on the server. The logic
    // for whether the Flush RPC is accepted is the same as that which is
    // implemented for Modify. In summary:
    //
    //   - if the election ID is equal to the previous highest known ID, the Flush
    //     is accepted.
    //   - if the election ID is greater than the previous highest known ID, the
    //     Flush is accepted.
    //   - if the election ID is less than the previous highest known ID, the
    //     Flush is rejected and an error with reason NOT_MASTER is returned.
    //
    // If the server is currently operating in ALL_PRIMARY mode the server should
    // return an error specifying ELECTION_ID_IN_ALL_PRIMARY reason.
```